src/Raven.Server/Documents/Indexes/Static/IndexCompilationCache.cs

```diff
@@ -50,7 +50,12 @@ public static StaticIndexBase GetIndexInstance(IndexDefinition definition, Raven
 
             try
             {
-                return result.Value;
+                var index = result.Value;
+                if (index is JavaScriptIndex javaScriptIndex)
+                {
+                    javaScriptIndex.TryRemoveFromCache = () => { IndexCache.TryRemove(key, out _); };
```

Comment:
IndexCache is static, what if I will have 2 identical indexes e.g. in two separate databases with sample data?

Reply:
https://github.com/ravendb/ravendb/blob/v4.2/src/Raven.Server/Documents/Indexes/Static/IndexCompilationCache.cs#L33

Reply:
Ok, didn't know that